packages/core/src/layout/ContentHeader/ContentHeader.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * TODO favoriteable capability
+ */
+
+import React, { ComponentType, Fragment, FC } from 'react';
+import { Typography, makeStyles } from '@material-ui/core';
+import Helmet from 'react-helmet';
+import { BackstageTheme } from '../../theme/theme';
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  container: {
+    width: '100%',
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+  },
+  leftItemsBox: {
+    flex: '1 1 auto',
+    marginBottom: theme.spacing(1),
+    minWidth: 0,
+    overflow: 'visible',
+  },
+  rightItemsBox: {
+    flex: '0 1 auto',
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    alignItems: 'center',
+    marginLeft: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    minWidth: 0,
+    overflow: 'visible',
+  },
+  description: {},
+  title: {
+    display: 'inline-flex',
+  },
+}));
+
+type DefaultTitleProps = {
+  title?: string;
+  className: string;
+};
+
+const DefaultTitle: FC<DefaultTitleProps> = ({
+  title = 'Unknown page',
+  className,
+}) => (
+  <Typography variant="h4" className={className} data-testid="header-title">
+    {title}
+  </Typography>
+);
+
+type ContentHeaderProps = {
+  titleComponent?: ComponentType;
+  description?: string;
+} & Pick<DefaultTitleProps, 'title'>;
```

Comment:
Think I'd prefer the ´title: DefatultTitleProps['title']´ approach, afaik that displays better in error messages and editor popups. Not a blocker though.

Reply:
I updated this type to use the other approach. 
But doing this, you lose the composition of the types of typescript, which could be redundant if you have more props to pick. Even here, `title?` is repeated 2 times, which is more prone to errors (like forgetting the `?` comma)